third_party/libusb/webport/src/libusb_js_proxy_unittest.cc

```diff
@@ -463,6 +464,134 @@ TEST_P(LibusbJsProxyTest, DeviceRefUnref) {
+                                                    .Add("direction", "out")
+                                                    .Add("type", "bulk")
+                                                    .Add("maxPacketSize", 64)
+                                                    .Get())
+                                           .Add(DictValueBuilder()
+                                                    .Add("endpointAddress", 130)
+                                                    .Add("direction", "in")
+                                                    .Add("type", "bulk")
+                                                    .Add("maxPacketSize", 64)
+                                                    .Get())
+                                           .Add(DictValueBuilder()
+                                                    .Add("endpointAddress", 131)
+                                                    .Add("direction", "in")
+                                                    .Add("type", "interrupt")
+                                                    .Add("maxPacketSize", 16)
+                                                    .Get())
+                                           .Get())
+                                  .Get())
+                          .Get())
+                  .Get())
+          .Get());
+
+  // Act.
+  libusb_config_descriptor* descriptor = nullptr;
+  EXPECT_EQ(libusb()->LibusbGetActiveConfigDescriptor(devices[0], &descriptor),
+            LIBUSB_SUCCESS);
+
+  // Assert.
+  ASSERT_TRUE(descriptor);
+  EXPECT_EQ(descriptor->bLength, sizeof(libusb_config_descriptor));
+  EXPECT_EQ(descriptor->bDescriptorType, LIBUSB_DT_CONFIG);
+  EXPECT_EQ(descriptor->wTotalLength, sizeof(libusb_config_descriptor));
+  EXPECT_EQ(descriptor->bConfigurationValue, 1);
+  EXPECT_EQ(descriptor->bNumInterfaces, 1);
+  EXPECT_EQ(descriptor->extra, nullptr);
+  EXPECT_EQ(descriptor->extra_length, 0);
+  const auto& interface = descriptor->interface[0];
+  EXPECT_EQ(interface.num_altsetting, 1);
+  const auto& interface_descriptor = interface.altsetting[0];
+  EXPECT_EQ(interface_descriptor.bLength, sizeof(libusb_interface_descriptor));
+  EXPECT_EQ(interface_descriptor.bDescriptorType, LIBUSB_DT_INTERFACE);
+  EXPECT_EQ(interface_descriptor.bInterfaceNumber, 0);
+  ASSERT_EQ(interface_descriptor.bNumEndpoints, 3);
+  EXPECT_EQ(interface_descriptor.bInterfaceClass, 11);
+  EXPECT_EQ(interface_descriptor.bInterfaceSubClass, 0);
+  EXPECT_EQ(interface_descriptor.bInterfaceProtocol, 0);
+  ASSERT_TRUE(interface_descriptor.extra);
+  ASSERT_GT(interface_descriptor.extra_length, 0);
+  EXPECT_EQ(std::vector<uint8_t>(
+                interface_descriptor.extra,
+                interface_descriptor.extra + interface_descriptor.extra_length),
+            kInterfaceExtraData);
+  const auto* endpoint = interface_descriptor.endpoint;
+  ASSERT_TRUE(endpoint);
```

Comment:
optional nit: You could also test the length of this array

Reply:
It's tested above when checking the `bNumEndpoints`, but as it was hard to spot (even for myself, next day after I wrote this code :-) ) I've moved it close to this line.

Reply:
No, I did see that. But I thought that buggy code could still produce an array that is larger (or smaller, though this would be caught implicitly in the assertions below). I don't expect that this would lead to incorrect behavior though, so I considered this optional.

Reply:
Got it. With C-style dynamic arrays it's almost impossible to check it properly though. At least our ASan tests would catch a too-small array, but a too-big one wouldn't be caught.